mvc/models.js

```diff
@@ -33,18 +33,19 @@ exports.getTopicsModel = () => {
     })
 }
 
-exports.getArticlesModel = () => {
-    return db
-        .query(
-            `SELECT articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, CAST(COUNT(comments.article_id) AS INT) AS comment_count
+exports.getArticlesModel = (sort_by, order) => {
+    let sortByValue
+    sort_by ? (sortByValue = sort_by) : (sortByValue = "created_at")
+    let orderValue
+    order ? (orderValue = order) : (orderValue = " DESC")
+    let query = `SELECT articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, CAST(COUNT(comments.article_id) AS INT) AS comment_count
         FROM articles
         LEFT JOIN comments ON articles.article_id = comments.article_id
         GROUP BY articles.article_id
-        ORDER BY created_at DESC;`
-        )
-        .then(({ rows }) => {
-            return rows
-        })
+        ORDER BY ${sortByValue} ${orderValue}`
```

Comment:
This is currently open to SQL injection. These values could be something malicious as you have no validation in your model to prevent it. Take a look at how we can use a greenlist array to verify it beforehand 👍